File: 2025-1/sesion-de-clase-17_1/Persona.cpp
/* 
 * File:   Persona.cpp
 * Author: erichuiza
 * 
 * Created on May 26, 2025, 4:29 PM
 */

#include "Persona.h"

Persona::Persona() {
}

Persona::Persona(int dni, const string nombre) {
    this->dni = dni;
    this->setNombre(nombre);
}

Persona::Persona(const Persona& orig) {
    *this = orig;
}

int Persona::getDni() const {
    return dni;
}

void Persona::setDni(int dni) {
    this->dni = dni;
}

string Persona::getNombre() const {
    return this->nombre;
}

void Persona::setNombre(const string nombre) {
    this->nombre = nombre;
}

bool Persona::cargarDatos(ifstream& archivo) {
    int dni;
    if (!archivo.eof() && archivo >> dni) {
        archivo.ignore();
        char *nombre = leerCadena(archivo);
        
        this->setDni(dni);
        this->setNombre(nombre);
        
        return true;
    }

    return false;
}

bool operator>>(ifstream& archivo, Persona& persona) {
    return persona.cargarDatos(archivo);
}

ostream& operator<<(ostream& os, const Persona& persona) {
    persona.imprimir(os);
    return os;
}

Persona& Persona::operator=(const Persona& persona) {
    this->setDni(persona.getDni());
    this->setNombre(persona.getNombre());
    
    return *this;
}

void Persona::imprimir(ostream& os) const {
    os << left << setw(15) << this->dni
        << setw(50) << this->nombre;
}

Persona::~Persona() {
}

File: 2025-1/sesion-de-clase-22/src/Persona.java
/**
 *
 * @author erichuiza
 */
public class Persona {
    private int dni;
    private String nombre;
    private int edad;
    private String apellido;
    
    public Persona() {
    }
    
    public Persona(int dni, String nombre, int edad) {
        this.dni = dni;
        this.nombre = nombre;
        this.edad = edad;
    }
    
    public int getDni() {
        return this.dni;   
    }
    
    public void setId(int dni) {
        this.dni = dni;
    }
    
    public String getNombre() {
        return this.nombre;
    }
    
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public int getEdad() {
        return this.edad;
    }
    
    public void setEdad(int edad) {
        this.edad = edad;
    }
    
    public void imprimir() {
        System.out.println("DNI: " + dni);
        System.out.println("Nombre: " + nombre);
        System.out.println("Edad: " + edad);
    }

    @Override
    public String toString() {
        String str = "DNI: " + dni + "\n";
        str += "Nombre: " + nombre + "\n";
        str += "Edad: " + edad + "\n";
        return str;
    }
    
    
}
